source/auth/tests/connection-string.json

```diff
@@ -376,6 +376,63 @@
       "description": "should throw an exception if no username/password provided (userinfo implies default mechanism)",
       "uri": "mongodb://:@localhost.com/",
       "valid": false
+    },
+    {
+      "description": "should recognise the mechanism (MONGODB-AWS)",
+      "uri": "mongodb://localhost/?authMechanism=MONGODB-AWS",
+      "hosts": null,
+      "valid": true,
+      "warning": false,
+      "auth": {
+        "username": null,
+        "password": null,
+        "db": "$external"
+      },
+      "options": {
+        "authmechanism": "MONGODB-AWS"
+      }
+    },
+    {
+      "description": "should throw an exception if username and no password (MONGODB-AWS)",
+      "uri": "mongodb://user@localhost/?authMechanism=MONGODB-AWS",
+      "hosts": null,
+      "valid": false,
+      "warning": false,
+      "auth": null,
+      "options": null
+    },
+    {
+      "description": "should use username and password if specified (MONGODB-AWS)",
+      "uri": "mongodb://user:password@localhost/?authMechanism=MONGODB-AWS",
+      "hosts": null,
+      "valid": true,
+      "warning": false,
+      "auth": {
+        "username": "user",
+        "password": "password",
+        "db": "$external"
+      },
+      "options": {
+        "authmechanism": "MONGODB-AWS"
+      }
+    },
+    {
+      "description": "should use username, password and session token if specified (MONGODB-AWS)",
+      "uri": "mongodb://user:password@localhost/?authMechanism=MONGODB-AWS&authMechanismProperties=AWS_SESSIONS_TOKEN:token",
```

Comment:
Can we add some URL encoded characters to this TOKEN to make sure drivers decode in the right place?

Perhaps `AWS_SESSIONS_TOKEN:token%21%40%23%24%25%5E%26%2A%28%29_%2Betc` which decodes to `token!@#$%^&*()_+etc`. I had to fix a bug in Python which would have been easier to find with this kind of test.

Reply:
Also the `authMechanismProperties` should be `AWS_SESSION_TOKEN` not `AWS_SESSIONS_TOKEN`.

Reply:
Good idea. Added it for the username and password in a separate test as well.